source/deploy-strategies.rst

```diff
@@ -0,0 +1,162 @@
+  bugs. It can be tricky figuring out a version of Salt that works.
+* Resource usage - the Salt master process seems to grow over time to
+  use a lot of memory on the server, even though it shouldn't be doing
+  anything at all when we're not deploying. This ties up memory our
+  site could be using.
+
+We are therefore moving away from Margarita for new projects.
+
+Tequila
+-------
+
+Tequila (see `tequila-*` projects on Caktus github)
+is a successor project to Margarita, which uses
+`Ansible <http://docs.ansible.com/>`_
+instead of Salt. It is hoped that Tequila will avoid some of the problems
+of Margarita.
+
+* Ansible error messages are much clearer, making debugging deploy problems
+  easier.
+* Ansible is agent-less, so when not deploying, it takes no resources on
+  the servers.
+* Ansible seems (so far) to have a better track record of not introducing
+  regressions with its updates.
+
+At present, Tequila is in a beta state, with several newer projects trying
+it out.
+
+Elastic Beanstalk
+-----------------
+
+`Elastic Beanstalk <https://aws.amazon.com/elasticbeanstalk/>`_
+is an AWS service that provides a nice front-end wrapper
+around some of their other services, including Elastic Load balancer and
+autoscaling. A `Caktus blog post <https://www.caktusgroup.com/blog/2017/03/23/hosting-django-sites-amazon-elastic-beanstalk/>`_ gave a quick overview.
+
+An advantage of EB over Margarita, Tequila, and FabulAWS is that it provides
+a higher level of abstraction, so developers don't have to concern themselves
+with as many details about the deploy when they just want something that works.
+
+We inherited one project that was already using Elastic Beanstalk and have
+had good experiences with it. It was a big enough project that we were grateful
+for the horizontal scaling (multiple web servers behind a load balancer), though
+I don't think we have needed autoscaling.
+
+Elastic Beanstalk is tied to AWS.
+
+Heroku
+------
+
+We've had a few projects use `Heroku <https://heroku.com>`_.
+Heroku is a well-designed and robust service
+that makes it easy to deploy projects to the Internet. It provides for easy
+deploy and scaling like Elastic Beanstalk. The only drawback is that running a
+site on Heroku incurs a premium cost compared to the run-time costs of most of
+these other alternatives. Of course, that does not take into account the high
+cost of our own labor in developing and maintaining some of these other solutions.
+Still, we haven't used Heroku much. Mostly we use it for quick prototypes, or
+if a client specifically requests it.
+
+Dokku
+-----
+
+`Dokku <http://dokku.viewdocs.io/dokku/>`_
+is an open-source alternative to Heroku. It takes the same approach to
+deploys, but doesn't (yet) offer all the services of Heroku. We are currently
+trialing Dokku with some internal projects and one client project.
+
+Dokku appears best suited to smaller projects, ones where we'd be comfortable
+running on a single server.
```

Comment:
I think it's important to mention that (IIUC) a major reason for this is that scaling and redundancy are difficult with Dokku, as it currently stands. If we had good approaches to those, then we might be happy to use Dokku on a larger project. (Or if my understanding is not accurate, then a small note explaining why Dokku is limited to small projects)